src/pudl/transform/nrelatb.py

```diff
@@ -0,0 +1,300 @@
+"""Transform NREL ATB data into well normalized, cleaned tables."""
+
+from typing import Self
+
+import pandas as pd
+from dagster import asset
+from pydantic import BaseModel, field_validator
+
+import pudl.helpers as helpers
+
+IDX_ALL = [
+    "report_year",
+    "mystery_code",
+    "core_metric_case",
+    "core_metric_variable_year",
+    "cost_recovery_period_years",
+    "scenario_atb",
+    "technology_description",
+    "technology_description_detail_1",
+    "technology_description_detail_2",
+    "core_metric_parameter",
+]
+"""Expected primary key columns for the raw nrelatb data.
+
+The normalized core tables we are trying to build will result in a subset of these columns.
+"""
+
+
+class TableUnstacker(BaseModel):
+    """Info needed to unstack a portion of the NREL ATB table."""
+
+    idx: list[str]
+    params: list[str]
+    """Values from the ``core_metric_parameter`` column to be included in this unstack."""
+
+    @field_validator("idx")
+    @classmethod
+    def idx_are_same_or_subset_of_idx_all(cls: Self, idx: list[str]):
+        """Are the :attr:`idx` columns either the same as or a subset of :attr:`IDX_ALL`?"""
+        assert set(idx).issubset(IDX_ALL)
+        return idx
+
+    @property
+    def idx_unstacked(self: Self) -> list[str]:
+        """Primary key columns after the table is unstacked.
+
+        All of the columns in :attr:`idx` except core_metric_parameter.
+        """
+        return [c for c in self.idx if c != "core_metric_parameter"]
+
+
+class Unstacker(BaseModel):
+    """Class that defines how to unstack the :func:`transform_start` table into all of the tidy NREL tables."""
+
+    rate_table: TableUnstacker
+    scenario_table: TableUnstacker
+    tech_detail_table: TableUnstacker
+
+    @property
+    def params_all(self: Self) -> list[str]:
+        """Compilation of all of the parameter values from each of the tables.
+
+        Also check if there are no duplicate params. We expect all of the parameters
+        across the :class:`TableUnstacker` to be unique.
+        """
+        params_all = (
+            self.rate_table.params
+            + self.scenario_table.params
+            + self.tech_detail_table.params
+        )
+        assert not {x for x in params_all if params_all.count(x) > 1}
+        return params_all
+
+
+def unstacker_of_tables() -> Unstacker:
```

Comment:
this is a compilation of parameters for how to unstack all of the three tables we want to break out from the raw rnel table. It feels kinda silly to make this into a function that just compiles parameters and use it for each of the individual tables, but it is helpful to run a few checks with the all three together (both pydantic validations and checks w/ the  `_core_nrelatb__transform_start` data).

this could be a global variable but I find it pretty frustrating ergonomically when trying to tweak parameters that get loaded & validated on import.

Reply:
removed this in favor of storing this info in `Unstacker` per @jdangerx 's suggestion